lib/rack/directory.rb

```diff
@@ -1,6 +1,11 @@
 # frozen_string_literal: true
 
 require 'time'
+require_relative '../rack'
+require_relative 'files'
+require_relative 'head'
+require_relative 'mime'
+require_relative 'utils'
```

Comment:
Here is an example of the slight awkwardness mentioned in **Primary Change** in the PR description. Technically the `require_relative '../rack'` is sufficient and the other 4 `require_relative`s are unnecessary. But personally, I kind of like seeing that `Rack::Directory` has a dependency on several other parts of Rack.

What do y'all think?

Reply:
I think being explicit is okay, but we should also avoid being unnecessarily explicit if it creates performance issues (i.e. lots of stats/hash lookups). Rack is loaded a lot.

Reply:
Totally fair!

Please double-check my understanding: wouldn't there still only be a one-time hit for each of these `require_relative`s? For example, once `lib/rack/directory.rb` was loaded - either via the `autoload` mechanism, the `require "rack/directory"` from a consuming app, or `require_relative` from w/in another part of Rack, any further attempts to require the file be a no-op since it's already loaded? Meaning these `require_relatives` wouldn't be invoked again, thus wouldn't cause extraneous hash lookups.